Build fail on Ubuntu 24.04 LTS

When building `svase` on Ubuntu 24.04 LTS the following build fails pops up (a build of the same script on 22.04 LTS works fine):
```
[ 40%] Building CXX object CMakeFiles/svase.dir/src/rewrite.cpp.o
In file included from /tmp/svase/deps/install/include/fmt/format.h:48,
                 from /tmp/svase/include/svase/util.h:10,
                 from /tmp/svase/include/svase/diag.h:10,
                 from /tmp/svase/include/svase/rewrite.h:11,
                 from /tmp/svase/src/rewrite.cpp:11:
/tmp/svase/deps/install/include/fmt/core.h: In instantiation of ‘constexpr fmt::v9::detail::value<Context> fmt::v9::detail::make_value(T&&) [with Context = fmt::v9::basic_format_context<fmt::v9::appender, char>; T = fmt::v9::join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >, char>&]’:
/tmp/svase/deps/install/include/fmt/core.h:1777:29:   required from ‘constexpr fmt::v9::detail::value<Context> fmt::v9::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v9::basic_format_context<fmt::v9::appender, char>; type <anonymous> = fmt::v9::detail::type::custom_type; T = fmt::v9::join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >, char>&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]’
/tmp/svase/deps/install/include/fmt/core.h:1901:77:   required from ‘constexpr fmt::v9::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, fmt::v9::join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, char>&}; Context = fmt::v9::basic_format_context<fmt::v9::appender, char>; Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, fmt::v9::join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, char>}]’
/tmp/svase/deps/install/include/fmt/core.h:1918:31:   required from ‘constexpr fmt::v9::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<Args>::type>::type ...> fmt::v9::make_format_args(Args&& ...) [with Context = basic_format_context<appender, char>; Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, char>&}]’
/tmp/svase/deps/install/include/fmt/core.h:3206:44:   required from ‘std::string fmt::v9::format(format_string<T ...>, T&& ...) [with T = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, char>}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, char> >]’
/tmp/svase/src/rewrite.cpp:298:18:   required from here
/tmp/svase/deps/install/include/fmt/core.h:1735:15: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
 1735 |   const auto& arg = arg_mapper<Context>().map(FMT_FORWARD(val));
      |               ^~~
/tmp/svase/deps/install/include/fmt/core.h:1735:46: note: the temporary was destroyed at the end of the full expression ‘fmt::v9::detail::arg_mapper<fmt::v9::basic_format_context<fmt::v9::appender, char> >().fmt::v9::detail::arg_mapper<fmt::v9::basic_format_context<fmt::v9::appender, char> >::map<fmt::v9::join_view<__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >, __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >, char>&>((* & val))’
 1735 |   const auto& arg = arg_mapper<Context>().map(FMT_FORWARD(val));
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/svase.dir/build.make:90: CMakeFiles/svase.dir/src/rewrite.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/svase.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -86,7 +86,7 @@ jobs:
         export GID=$(id -g)
         export REPO_PATH=${{github.workspace}}/
         export CMAKE_BUILD_PORTABLE=1
-        docker-compose -f .github/docker/docker-compose.yml up
+        docker compose -f .github/docker/docker-compose.yml up
 
     - name: Cache dependencies
       uses: actions/cache@v4

--- a/README.md
+++ b/README.md
@@ -60,7 +60,9 @@ The following requirements should be sufficient:
 - C++20 compatible compiler (GCC 10, Clang 16, XCode 14.3)
 - Python 3
 
-The only tested configuration is GCC 11.2.0 with CMake 3.20.2 and Python 3.6.
+The tested configurations are:
+- GCC 11.2, CMake 3.20 and Python 3.6.
+- GCC 13.1, CMake 3.30 and Python 3.8.
 Depending on where/how you installed the compiler, CMake may not find the correct one. In this case you can manually specify a compiler, `scripts/svase.env` provides and example for this.
 
 #### Build

--- a/src/driver.cpp
+++ b/src/driver.cpp
@@ -132,10 +132,11 @@ int driverMain(int argc, char **argv) {
   std::string slangArgs = cmdOptsRes.count("slang-args")
                               ? cmdOptsRes["slang-args"].as<std::string>()
                               : "";
-  auto slangCmd = fmt::format(
-      "{} {} {} --top {} {}", argv[0], slangArgs, builtinFlags,
-      cmdOptsRes["top"].as<std::string>(),
+  auto filesStr = fmt::to_string(
       fmt::join(cmdOptsRes["files"].as<std::vector<std::string>>(), " "));
+  auto slangCmd =
+      fmt::format("{} {} {} --top {} {}", argv[0], slangArgs, builtinFlags,
+                  cmdOptsRes["top"].as<std::string>(), filesStr);
   ok &= slangDriver.parseCommandLine(slangCmd);
   ok &= slangDriver.processOptions();
   diag.registerEngine(&slangDriver.sourceManager);

--- a/src/rewrite.cpp
+++ b/src/rewrite.cpp
@@ -294,9 +294,9 @@ void ParameterRewriter::handle(const TypeParameterDeclarationSyntax &pd) {
     auto assignSyn = TypeAssignmentSyntax(decl->name, &newEquals);
     newDeclStrs.emplace_back(assignSyn.toString());
   }
-  auto declStr =
-      fmt::format("{}{}{}", pd.keyword.toString(), pd.typeKeyword.toString(),
-                  fmt::join(newDeclStrs, ", "));
+  auto newDeclStr = fmt::to_string(fmt::join(newDeclStrs, ", "));
+  auto declStr = fmt::format("{}{}{}", pd.keyword.toString(),
+                             pd.typeKeyword.toString(), newDeclStr);
   if (pd.keyword.toString().empty()) {
     diag.log(
         DiagSev::Warning,
@@ -344,8 +344,9 @@ void ParameterRewriter::handle(const ParameterDeclarationSyntax &pd) {
   }
   // Todo: figure out keyword, if the last keyword in the parameter port list
   // was a localparam, it is localparam, otherwise it is parameter
+  auto newDeclStr = fmt::to_string(fmt::join(newDeclStrs, ", "));
   auto declStr = fmt::format("{}{}{}", pd.keyword.toString(),
-                             pd.type->toString(), fmt::join(newDeclStrs, ", "));
+                             pd.type->toString(), newDeclStr);
   if (pd.keyword.toString().empty()) {
     diag.log(
         DiagSev::Warning,

